src/LiveComponent/src/EventListener/LiveUrlSubscriber.php

```diff
@@ -68,34 +65,73 @@ public static function getSubscribedEvents(): array
     }
 
     /**
-     * @return array{
-     *     path: array<string, mixed>,
-     *     query: array<string, mixed>
-     * }
+     * @return array{ array<string, mixed>, array<string, mixed> }
      */
-    private function getLiveProps(MountedComponent $mounted): array
+    private function extractUrlLiveProps(MountedComponent $mounted): array
     {
-        $metadata = $this->metadataFactory->getMetadata($mounted->getName());
+        $pathProps = $queryProps = [];
 
-        $dehydratedProps = $this->liveComponentHydrator->dehydrate(
-            $mounted->getComponent(),
-            $mounted->getAttributes(),
-            $metadata
-        );
+        $mountedMetadata = $this->metadataFactory->getMetadata($mounted->getName());
 
-        $values = $dehydratedProps->getProps();
+        $dehydratedProps = $this->liveComponentHydrator->dehydrate($mounted->getComponent(), $mounted->getAttributes(), $mountedMetadata);
+        $props = $dehydratedProps->getProps();
 
-        $urlLiveProps = [
-            'path' => [],
-            'query' => [],
-        ];
+        foreach ($mountedMetadata->getAllUrlMappings($mounted->getComponent()) as $name => $urlMapping) {
+            if (array_key_exists($name, $props)) {
+                if ($urlMapping->mapPath) {
+                    $pathProps[$urlMapping->as ?? $name] = $props[$name];
+                } else {
+                    $queryProps[$urlMapping->as ?? $name] = $props[$name];
+                }
+            }
+        }
+
+        return [$pathProps, $queryProps];
+    }
 
-        foreach ($metadata->getAllUrlMappings($mounted->getComponent()) as $name => $urlMapping) {
-            if (isset($values[$name]) && $urlMapping) {
-                $urlLiveProps[$urlMapping->mapPath ? 'path' : 'query'][$urlMapping->as ?? $name] = $values[$name];
+    private function generateNewLiveUrl(string $previousUrl, array $pathProps, array $queryProps): string
+    {
+        $previousUrlParsed = parse_url($previousUrl);
+        $newUrl = $previousUrlParsed['path'];
+        $newQueryString = $previousUrlParsed['query'] ?? '';
+
+        if ([] !== $pathProps) {
+            $context = $this->router->getContext();
+            try {
+                // Create a temporary RouterContext pointing to the current controller
+                $tmpContext = clone $context;
+                $tmpContext->setMethod('GET');
+                $this->router->setContext($tmpContext);
+
+                $matched = $this->router->match($previousUrlParsed['path']);
+
+                $route = $matched['_route'];
+                unset($matched['_route']);
+
+                $parameters = $matched;
+                foreach ($pathProps as $name => $value) {
+                    if ($value) {
+                        $parameters[$name] = $value;
+                    }
+                }
```

Comment:
I'm still not sure about this part, I need to double-check if my tests data were wrong or not. If that's the case `$pathProps + $matched` should be enough